# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerScope.cpp
# Commit: df960f6a3f76
# Full Hash: df960f6a3f76a3fc92fe57054830649c15f4ff3a
# Author: Frédéric Wang <fwang@igalia.com>
# Date: 2025-01-17 21:57:54
# Regressor Bug: 1901492
# File Overlap Count: 1
# Description:
#   Bug 1901492 - Make GetTrustedTypesCompliantString work with workers. r=smaug,tschuster
#   
#   Make sure GetTrustedTypesCompliantString() work when the specified
#   global object is a WorkerGlobalScope. If that happens the function
#   is executed from the worker thread. The following changes are made:
# ==============================================================================

diff -r c6296179fcb4 -r df960f6a3f76 dom/workers/WorkerScope.cpp
--- a/dom/workers/WorkerScope.cpp	Fri Jan 17 13:35:40 2025 +0000
+++ b/dom/workers/WorkerScope.cpp	Fri Jan 17 13:35:40 2025 +0000
@@ -930,6 +930,7 @@
 }
 
 TrustedTypePolicyFactory* WorkerGlobalScope::TrustedTypes() {
+  AssertIsOnWorkerThread();
   if (!mTrustedTypePolicyFactory) {
     mTrustedTypePolicyFactory = MakeRefPtr<TrustedTypePolicyFactory>(this);
   }